src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/mapTableToJson.test.ts

```diff
@@ -513,6 +515,136 @@ describe('mapTableToJson', () => {
     ]);
   });
 
+  test('correctly adds groups across three levels of column headers with same labels', () => {
+    const result = mapTableToJson({
+      tableHeadersConfig: testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabelsConfig,
+      subjectMeta:
+        testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabels.subjectMeta,
+      results:
+        testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabels.results,
+    }).tableJson;
+
+    expect(result.thead).toEqual([
```

Comment:
Should we add a generic type to `toEqual` here in order to get more precise compile-time feedback? (similar to the rest of the test)